File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.py
class Solution(object):
    def reverseDegree(self, s):
        total = 0
        
        for i, char in enumerate(s):
            reversed_value = ord('z') - ord(char) + 1
            position = i + 1
            
            total += reversed_value * position
        
        return total

File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.cpp
class Solution {
public:
    int reverseDegree(string s) {
        int total = 0;
        for (int i = 0; i < s.length(); ++i) {
            char c = s[i];
            int reverse_val = 'z' - c + 1; 
            int position = i + 1;        
            total += reverse_val * position;
        }
        return total;
    }
};
